log/log.go

```diff
@@ -0,0 +1,43 @@
+package log
+
+import (
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var Gl *zap.Logger
+
+func InitializeLogger() *zap.Logger {
+
+	config := zap.NewProductionEncoderConfig()
+	config.EncodeTime = zapcore.ISO8601TimeEncoder
+
+	fileEncoder := zapcore.NewJSONEncoder(config)
+	consoleEncoder := zapcore.NewConsoleEncoder(config)
+
+	logFile, _ := os.OpenFile("log.json", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	writer := zapcore.AddSync(logFile)
+	defaultLogLevel := zapcore.DebugLevel
+	core := zapcore.NewTee(
+		zapcore.NewCore(fileEncoder, writer, defaultLogLevel),
+		zapcore.NewCore(consoleEncoder, zapcore.AddSync(os.Stdout), defaultLogLevel),
+	)
+
+	Gl = zap.New(core, zap.AddCaller(), zap.AddStacktrace(zapcore.ErrorLevel))
+	return Gl
+
+}
+
+func LogInfo(message string) {
+	if Gl != nil {
+		Gl.Info(message)
+	}
+}
+
+func LogError(err error) {
+	if Gl != nil {
+		Gl.Error("Error occurred: ", zap.Error(err))
+	}
+}
```

Comment:
We don't need this abstraction functions, use `log.Gl` directly.

Reply:
fixed for categories.go file edition